core/js/src/typespecs/prompt.ts

```diff
@@ -0,0 +1,122 @@
+import { z } from "zod";
+
+export const roleSchema = z.enum([
+  "system",
```

Comment:
Are these roles supposed to align with the RBAC roles? If it's prompt-specific, maybe `promptRole` to avoid naming confusion?